archive_feedstock.py

```diff
@@ -28,57 +38,108 @@ def archive_repo(owner, repo, archive=True, check_only=False):
     )
     raise_json_for_status(r)
 
-    target_status = f"{'' if archive else 'un'}archived"
+    if task == "archive":
+        target_status = "archived"
+    else:
+        target_status = "unarchived"
+
     data = r.json()
-    if data["archived"] == archive:
-        raise RuntimeError(f"{owner}/{repo} is already {target_status}!")
-    elif check_only:
+    if task == "archive" and data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
+        return
+
+    if task == "unarchive" and not data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
         return
 
     r = requests.patch(
         f"https://api.github.com/repos/{owner}/{repo}",
         headers=headers,
-        json={"archived": archive}
+        json={"archived": True if task == "archive" else False}
     )
     raise_json_for_status(r)
 
-
-def feedstocks(directory="archive"):
-    for path in Path(directory).glob("*.txt"):
-        if path.name == "example.txt":
-            continue
-        with open(path) as f:
-            for line in f:
-                line = line.strip()
-                if line and not line.startswith("#"):
-                    yield line
+    print("feedstock %s was %s" % (repo, target_status), flush=True)
 
 
-def main(owner="conda-forge", check_only=False):
-    exceptions = []
-    for task in "archive", "unarchive":
-        seen = set()
-        for feedstock in feedstocks(task):
-            verb = "Archiving" if task == "archive" else "Unarchiving"
-            print(f"{verb} {owner}/{feedstock}-feedstock...", end=" ")
-            if feedstock in seen:
-                print("[SKIP]")
+def process_feedstocks_in_file(task_file, task):
+    pkgs_to_do_again = []
+    with open(task_file, "r") as fp:
+        for line in fp.readlines():
+            line = line.strip()
+            if line.startswith("#") or len(line) == 0:
                 continue
-            seen.add(feedstock.lower())
+
             try:
-                archive_repo(owner, f"{feedstock}-feedstock", archive=task == "archive", check_only=check_only)
-            except Exception as exc:
-                exceptions.append((feedstock, exc))
-                print("[!!]")
+                process_repo(line, task)
+            except Exception as e:
+                print(
+                    "failed to %s '%s': %s" % (task, line, repr(e)),
+                    flush=True,
+                )
+                pkgs_to_do_again.append(line)
+
+    if pkgs_to_do_again:
+        with open(task_file, "w") as fp:
+            fp.write(
+                "# %s failed for these feedstocks - "
+                "trying again later\n" % task
+            )
+            for pkg in pkgs_to_do_again:
+                fp.write(pkg + "\n")
+        subprocess.check_call(f"git add {task_file}", shell=True)
+        subprocess.check_call(
+            f"git commit --allow-empty -m 'Keeping {task_file} "
```

Comment:
```suggestion
            f"git commit -m 'Keeping {task_file} "
```

It doesn't have to be empty, right? Or is it just in case there were no modifications in the end? 👀

Reply:
There was some bug in the token resets code which required this line: https://github.com/conda-forge/admin-requests/commit/1bb65709e47b0448ab822e82605c6c2f433a8fcd

Let's keep it for now.